### PR TITLE
Fix sidebar dropdowns

### PR DESCRIPTION
**Pull request by wmcconnell**
## Summary
Changed the way we were dealing with conditionally displaying the sidebar on three, specific, archive pages:

* archive.php
* category.php
* single.php

There was some code on these three pages that would modify the basic page layout if the sidebar was empty - allowing the archived posts to fill the available space. The method in place was to use output buffering to capture the sidebar, test to see if it was an empty string, and modify the classes applied to certain `<div>` elements based on that test.

If the sidebar was **not** empty, we echoed it out after cleaning it up with the `wp_kses_post()` method. Unfortunately, passing the sidebar through that sanitization method removed HTML tags that are required for some built-in WordPress widgets.

I changed the logic to use the `is_active_sidebar()` method which will return a value of **false** if the sidebar is not active on the page, **or** if it has no registered widgets (essentially the same as it being empty). Using this built-in Wordpress method, and then outputting the sidebar via the standard `get_sidebar()` method should accomplish the same goal without breaking existing sidebar widgets.

## Notes

* Is there a situation where we expect to display posts (either **archived** or **by category**) without a sidebar? The more I dug around through the files, the less I thought that this situation would ever really happen. It applies only to displaying actual Wordpress **posts** - and only when filtering by category or date, or viewing a single post. It would not apply to our static pages - and would not seem to apply to our events or news pages, as they override the default sidebar anyway.

* The Bootstrap class we apply to the sidebar when it's empty is `hidden-xs`, which, for the record, is **only** going to hide the sidebar div on extra small screens. Of course, when Wordpress finds no sidebar widgets to render, it doesn't output anything regardless of screen size. Still, it's worth nothing if odd things show up in the future.

### CODE REVIEW

**Comment by nathan-rollins** [comment]
Re: Notes

1. It is unusual but we did display category posts _sans_ sidebar on an older design of the Sustainability Connect website.

2. Actually, hidden-xs operates on xs and all wider widths. This is true of all size modifiers in Bootstrap classes. The given size is the minimum viewport breakpoint the class is applied to.

File: single.php [hunk 1]
```diff
@@ -18,45 +18,42 @@
...
-                <?php the_title( '<h1 class="entry-title">', '</h1>' ); ?>
-                <div class="entry-meta">
-                  <?php asu_webstandards_posted_on(); ?>
-                </div><!-- .entry-meta -->
-              </header><!-- .entry-header -->
-              <div class="single">
-                <?php
-                while ( have_posts() ) {
-                  the_post();
+          <?php
```
Comment (nathan-rollins) [inline single.php]: This block is a mismatch for the page. The Loop has been pasted in here, but this is a single post view. The original logic will need to be refactored just for your sidebar fix.

**Comment by nathan-rollins** [comment]
EDIT: Actually, Bootstrap v3 does only apply the hidden-* classes to the specified viewport size, and not to the larger sizes. Bootstrap v4 applies the hidden-* classes consistent with the other size classes as minimum breakpoints.

**Comment by wmcconnell** [comment]
I've reverted the parts of `single.php` that display the post. Turns out the original code also used a loop, even on a single-post page. This is consistent with the way that Wordpress does it; see [the 2017 Theme single.php](https://github.com/WordPress/WordPress/blob/master/wp-content/themes/twentyseventeen/single.php).

I'm not sure what to do about the `hidden-xs` class being used. I've tested it, and it does hide the sidebar (even if there is content to render) on extra small screens, which appears to be what was originally intended. Still, it may be part of a larger discussion about how we want our theme to behave on smaller screens.

**Comment by wmcconnell** [comment]
Updated `index.php` to use the same sidebar displaying logic used by the other pages.

**Pull request merged**